### ISSUE TITLE
OpenTasks causes many syncs (draining battery) for seemingly no reason

### ISSUE THREAD

**Issue by stevesbrain** [etesync/android#86]
As per logs provided offline, seems OpenTasks causes lengthy/numerous syncs for no apparent reason. Persists even after complete remove and reinstall of OpenTasks. Doesn't occur every time, but does occur many of the syncs. Let me know if you need any more troubleshooting!

**Comment by tasn** [comment]
Damn you OpenTasks, why do you hate me? :|

The "happens after reinstall" is baffling. I wonder if maybe EteSync is triggering a change that's causing OpenTasks to trigger a change and then they infinitely loop.

What are your (if you don't mind sharing)
EteSync version?
OpenTasks version?
Android version? (Rom?)

Thanks!

**Comment by stevesbrain** [comment]
EteSync - 1.7.0
OpenTasks - 1.1.20
Android Version - Pixel 2XL running stock Android 9, build number PQ3A.190801.002 (when this started, I removed any customisations I had; which was basically just same ROM but with Magisk installed - I downloaded the stock ROM from Google, verified all checksums, reflashed everything, and re-locked phone after doing so, so it's basically an OOB experience)

Cheers

**Comment by tasn** [comment]
Thanks for testing it so thoroughly! Apparently I'm still on 1.1.17 (updating now), and there may have been relevant changes in 1.1.18. I'm upgrading and testing now.

**Comment by tasn** [comment]
Ah, I think I found it! I haven't managed to reproduce it, but looking at the OpenTasks code there sees to be a very relevant change! https://github.com/dmfs/opentasks/commit/1dcd94349446c76a4e79146c46b178a08d572b1d

Are you able to try OpenTasks 1.1.19 by any change? (You can get it from f-droid: https://f-droid.org/en/packages/org.dmfs.tasks/ )

**Comment by tasn** [comment]
I also updated some of our dependencies (one of which handles the OpenTasks integration). Here's a test version signed with the Google Play release key: https://stosb.com/~tom/opentasks-fix.apk

Please, if you can also test this that would be great. Thanks!

**Comment by stevesbrain** [comment]
Would you prefer me test 1.1.19 with current EteSync, 1.1.19 with the above EteSync, or above EteSync with 1.1.20? :)

**Comment by tasn** [comment]
Sorry for the confusion. :)

Please try the EteSync APK. If it still doesn't work, please also try downgrading to 1.1.19 (keeping the EteSync APK installed is fine).

**Comment by stevesbrain** [comment]
Will do now!

**Comment by stevesbrain** [comment]
Seems to have resolved the issue with your test build + latest tasks :) Will keep an eye on it, but good so far!

**Comment by tasn** [comment]
I'm feeling a combination of :angry: and :tada: 
Annoying that they broke backwards compat...

If you can, please report back in a few hours to make sure it's really solved, and if it is, I'll release a new version with the fix! :)

**Comment by stevesbrain** [comment]
So sad news - it took a while but eventually started the same issue again! Same symptoms - rapidly expanding log file, super battery drain, hot phone. Will keep the same EteSync version, but have reverted opentasks to 1.1.19. Will let you know how that goes!

**Comment by tasn** [comment]
:(

OK, let's see how it goes with 1.1.19. Thanks!

**Comment by stevesbrain** [comment]
Not dice - battery drain still present, and a couple of massive log files already :(

**Comment by tasn** [comment]
:(

Thanks for the update. One more idea: maybe try 1.1.17?
Do you have many tasks? Were they all created using OpenTasks or also other clients?

In parallel, I'll try to figure out how to extract more debug info to know what's going on. 
@dmfs: any idea what's going on? Essentially it looks like OpenTasks keeps on triggering sync requests non-stop. I've already seen it triggering double sync requests for changes before (so twice in parallel for the same change), but this looks like something else. Have you seen anything like this before? Any idea how to even begin to debug this? Thanks!

**Comment by stevesbrain** [comment]
I will try tomorrow (just heading to bed now) - I don't have many tasks though (6 in one, 4 in the other, about 20 journal entries each). All created in opentasks, one (now cleared job that exists in journal still) was edited in thunderbird a while back (via the dav client and carddav plugins). I could also try with a fresh journal if needed?

**Comment by tasn** [comment]
No rush. :)
Don't try with a fresh journal yet, because we don't want to lose the ability to reproduce the issue. Could you maybe try deleting the task you edited in thunderbird? Could you do it from thunderbird? After you do that, could you reinstall open tasks (so it doesn't have any cached data). I wonder if that task contains anything that causes opentasks to barf.

**Comment by stevesbrain** [comment]
Definitely still happens on 1.1.17, so I'll try figure out which of the tasks it was that I edited with thunderbird :P I did it more as a test than anything, so no real "association" to track it back with :P does the dav bridge store any logs that may help locate it? Otherwise I will try see if thunderbird does!

**Comment by tasn** [comment]
No logs, sorry.
You said you don't have that many events, maybe try deleting one by one? Or five by five? Normally events have a PRODID in them (which you can see in the raw EteSync journal log), but I don't know if PROID is updated when a task is merely edited.

**Comment by stevesbrain** [comment]
Hmm so by process of elimination, each successive tasks I deleted basically reduced, but did not eliminate battery drain (and resulted in progressively smaller log files). Completely removed tasks (but keeping existing journal) makes negligible sizes log files and seemingly no battery drain. Will try adding fresh entries now and see what happens.

**Comment by tasn** [comment]
Maybe try a fresh journal first (so create a new task journal) and then if that still fails, a fresh account.

I wonder why it's happening, but it looks like trying to sync any entries causes an infinite loop. I guess what happens is that OpenTasks notifies EteSync about a new task. EteSync marks it as "handled" which gets OpenTasks to trigger another sync which aborts the current sync which restarts this loop.
I wonder why is OpenTasks doing that, but I also wonder why do sync requests abort rather than queue.

**Comment by stevesbrain** [comment]
Have done so - first glance it appears new journal with new task has no excess battery usage, but after editing that task (ie completing it or part of it) it does begin the excess consumption. Will try again tomorrow to confirm if this is always the behaviour or if I just got lucky when adding the task!

**Comment by tasn** [comment]
Interesting. Yeah, please keep the information flowing. Thanks a lot for all your help!

**Comment by dmfs** [comment]
OpenTasks sends a sync trigger whenever a change was made by a non-sync-adapter app. You can see the relevant line here: https://github.com/dmfs/opentasks/blob/master/opentasks-provider/src/main/java/org/dmfs/provider/tasks/SQLiteContentProvider.java#L332
So if you see frequent syncs, please ensure that **every** operation sent by the sync adapter has the `CALLER_IS_SYNCADAPTER` parameter set to `true`.
Even tiny changes like setting a just sync field on a calendar will trigger a sync if the parameter is not present.
I think, I already have a story for not triggering a sync when a sync-adapter-only field has been changed, but that's not implemented yet.
The recent commit regarding the `PROVIDER_CHANGED` broadcast is not related to this issue (unless you're listening for that broadcast, which you shouldn't). Also that change has not been released yet.

**Comment by dmfs** [comment]
I've just checked the OpenTasks `git diff` since 1.1.17 and I couldn't spot any backwards compatibility breaking change. In addition I can say that a change of the [sync columns of a task list](https://github.com/dmfs/opentasks/blob/master/opentasks-contract/src/main/java/org/dmfs/tasks/contract/TaskContract.java#L437) should [not trigger a sync](https://github.com/dmfs/opentasks/blob/master/opentasks-provider/src/main/java/org/dmfs/provider/tasks/TaskProvider.java#L1023).

@stevesbrain do you use any other app which acts on task updates?

**Comment by stevesbrain** [comment]
@dmfs thanks for weighing in as well; I don't use any other apps that act on any task updates (nothing that reads them or creates them etc). In fact, in the list apps with permissions for "tasks", only EteSync exists (OpenTasks doesn't have or prompt for it, so I'm guessing it's not required?). Thanks!

**Comment by stevesbrain** [comment]
@tasn Yep, was able to replicate the same again today. New journal, new task; no battery drain for an hour or so (syncing every 15). Edited that task (added something to the "to do" list for the task) and drain started almost straight away (lost 5% in a few minutes with screen off etc). Anything further you want with that account, or want me to try with a fresh one?

**Comment by tasn** [comment]
@dmfs, what's interesting is that it works for everyone else (myself) included, and anyhow, we use the same libraries as DAVx5 for interacting with the tasks, so it's unlikely we are doing anything wrong.

Looking at [the code](https://github.com/etesync/android/blob/master/app/src/main/java/com/etesync/syncadapter/resource/LocalTask.kt) we only ever touch `SYNC1-3`, `_SYNC_ID` and `_DIRTY` (in addition to the content ones).

`CALLER_IS_SYNCADAPTER`: we use the same libraries as DAVx5, so it's already set for us.

Is there a way to get debug information from OpenTasks to know why it's requesting a sync?

**Comment by dmfs** [comment]
I've created a tiny app which monitors all content observer notifications issued by OpenTasks: https://github.com/dmfs/contentobserver
This app will print all Uris which OpenTasks passes to [resolver.notifyChange](https://github.com/dmfs/opentasks/blob/master/opentasks-provider/src/main/java/org/dmfs/provider/tasks/SQLiteContentProvider.java#L333). Unfortunately you won't see which of those have the `syncToNetwork` flag set, but it may indicate whether someone posts an operation without `CALLER_IS_SYNCADAPTER` flag.
Please note that the app only monitors changes while it is running (the app may survive short periods of background operation though).
Also note that changes may trigger more than one Uri at a time. That's intended behavior (e.g. updating a task causes a notification on the task Uri, the instances directory and the task directory).

Not all Uris lacking the `CALLER_IS_SYNCADAPTER` parameter will trigger a sync. This decision is made based on the Uri which was called initially, not the one which is notified.

Please note, I didn't implement the permission requests yet (this is really just a quick hack), so you'll have to grant the task read permissions manually in "Settings" -> "Apps & Notifications" -> "App permissions" (depending on your Android version). Without the permissions the app will probably just crash.

**Comment by tasn** [comment]
@stevesbrain, could you maybe try having this running while the sync goes crazy? Would be very interesting to see what it has to say.

Thanks!

**Comment by stevesbrain** [comment]
Sure thing @tasn - I'll run that tomorrow and see what we can see. Thanks @dmfs !

**Comment by stevesbrain** [comment]
The debug from ContentObserver:

`20190829T185259 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185259 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185300 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185302 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185302 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185302 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185307 content://org.dmfs.tasks/tasks/1?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185307 content://org.dmfs.tasks/instances
20190829T185307 content://org.dmfs.tasks/tasks
20190829T185318 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185319 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185319 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185443 content://org.dmfs.tasks/instances
20190829T185443 content://org.dmfs.tasks/tasks/1
20190829T185443 content://org.dmfs.tasks/tasks
20190829T185523 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185523 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185523 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185524 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185524 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185524 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185527 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185527 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185527 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185531 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185531 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185531 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185531 content://org.dmfs.tasks/tasks/1?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185531 content://org.dmfs.tasks/instances
20190829T185531 content://org.dmfs.tasks/tasks
20190829T185730 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185730 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T185730 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190134 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190134 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190134 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190200 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190200 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190200 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190205 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190205 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190205 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190238 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190238 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190238 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190241 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190241 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190241 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190247 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190247 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190247 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190256 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190256 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190256 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190303 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190303 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190303 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190305 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190305 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190305 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190342 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190342 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190342 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190458 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190458 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190458 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190512 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190512 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190512 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190519 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190519 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190519 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190522 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190522 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190522 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190551 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190551 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190551 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190622 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190622 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190622 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190622 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190622 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190622 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190639 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190639 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190639 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190650 content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190650 content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T190650 content://org.dmfs.tasks/tasklists/2?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
`

For the debug logs from EteSync from the same time period, I've added them [here](https://f001.backblazeb2.com/file/etesync-debug/debug.tar.gz.gpg) for you @tasn (using your GPG key).

**Comment by tasn** [comment]
Thanks @stevesbrain 

@dmfs, what do you make of it?

**Comment by dmfs** [comment]
@tasn do you use single updates or batch updates when you modify the task lists? I'm trying to reconstruct what's going on here. Which **task list**  columns do you update after a sync?

**Comment by tasn** [comment]
@dmfs: as he said, he's only changing one task item, so is batch even relevant?
What are all of those tasklists/2, 3 and 4? I think he only has one task list... Is this normal?

Columns: as said above:
Looking at [the code](https://github.com/etesync/android/blob/master/app/src/main/java/com/etesync/syncadapter/resource/LocalTask.kt) we only ever touch `SYNC1-3`, `_SYNC_ID` and `_DIRTY` (in addition to the content ones when the server changes things).

Other than that, we use the same library as DAVx5: https://gitlab.com/bitfireAT/ical4android/

**Comment by stevesbrain** [comment]
Just for clarity - I still have my original 2 task lists/journals (they are now empty). I think OpenTasks creates local one by default (which I didn't delete this time), and I created a 3rd EteSync journal (the only one I've been creating entries in). I can completely delete my other journals if need be - just hadn't yet in case we ended up needing to view one of those entries etc. Hope that explains (maybe) where the 4 are coming from?

**Comment by tasn** [comment]
Thanks for the clarification. The local one is not relevant, and this clarifies why you have 3 of them. Could you maybe delete the 2 journals? What happens if you have only one journal? Is it still broken for you? I wonder if that has anything to do with it, because that's a more unique scenario.

**Comment by stevesbrain** [comment]
Absolutely - shall try that out now!

**Comment by stevesbrain** [comment]
Same deal - still kills battery, even with only 1 journal. Content observer seems much more quiet (though this is only for 1 full sync):


`
20190829T214035 content://org.dmfs.tasks/instances
20190829T214035 content://org.dmfs.tasks/tasks
20190829T214035 content://org.dmfs.tasks/tasks/2
20190829T214056 content://org.dmfs.tasks/tasklists/5?account_name=ete%domain&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
20190829T214106 content://org.dmfs.tasks/tasklists/5?account_name=ete%domain&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
`

**Comment by dmfs** [comment]
> he said, he's only changing one task item, so is batch even relevant?

I'm just trying to understand which code path this took. Batch request and update requests take slightly different paths. FYI, the [class which ultimately triggers the update](https://github.com/dmfs/opentasks/blob/master/opentasks-provider/src/main/java/org/dmfs/provider/tasks/SQLiteContentProvider.java) is an almost unmodified copy of the original class in the Android sources.

> Columns: as said above:
Looking at the code we only ever touch SYNC1-3, _SYNC_ID and _DIRTY (in addition to the content ones when the server changes things).

Those are **task** sync columns, but we're talking about **list** sync columns here.

**Comment by tasn** [comment]
Sorry for the confusion!
The relevant file for the TaskList: https://github.com/etesync/android/blob/master/app/src/main/java/com/etesync/syncadapter/resource/LocalTaskList.kt

Looks like we are chanigng `NAME`, `LIST_COLOR`, `_SYNC_ID` and the code seems very similar to DAVx5 (we made some changes, but not a lot).

**Comment by dmfs** [comment]
I see. Looking at our code, I don't think we set the `syncToNetwork` flag when calling [notifyChange](https://developer.android.com/reference/android/content/ContentResolver.html#notifyChange(android.net.Uri,%20android.database.ContentObserver,%20boolean)). We should evaluate `caller_is_syncadapter` flag either [here](https://github.com/dmfs/opentasks/blob/master/opentasks-provider/src/main/java/org/dmfs/provider/tasks/SQLiteContentProvider.java#L212) or [here](https://github.com/dmfs/opentasks/blob/master/opentasks-provider/src/main/java/org/dmfs/provider/tasks/SQLiteContentProvider.java#L294) depending on whether you use batch operations or not.

That being said, the app does send the `PROVIDER_CHANGED` broadcast when you update the task list like that. Though, this broadcast should not trigger a sync. It is (was) meant for widgets and other apps which may need to take action when a content provider has been updated. Since you indeed update the name and color columns, the broadcast needs to be sent out (to inform listening clients of the new values).

I can provide a test version of OpenTasks to @stevesbrain which writes the arguments of the `notifyChange` call to the log, to ensure we don't really trigger a network sync. @stevesbrain would you want to try that out?

In addition I'll try to come up with a test case (and some design changes to allow for testing) on whether we set the flag appropriately.

@tasn As a workaround you could test if the name and color values are actually different. In general there is no value in updating a value to the same value again. It only causes workload for no reason. Even if we find the reason for this frequent sync issue and fix it, updating the value for no reason still causes updates to the UI and trigger broadcasts to listening apps although nothing has changed.

I've created dmfs/opentasks#836 to avoid database writes (and triggering broadcasts) on updates which don't update anything.

**Comment by stevesbrain** [comment]
@dmfs I am more than happy to try that out - just let me know when ready, and where to get it :)

**Comment by tasn** [comment]
@dmfs: thanks a lot! I can see what you're talking about and added the workaround. Isn't OpenTasks better suited to check if something have changed and only trigger a "changed" for the UI applications if that's the case?

@stevesbrain: could you please try: https://stosb.com/~tom/opentasks-fix.apk so we know if it fixes the issue? This is unrelated to @dmfs logging app which you'd need to check with the broken EteSync version.

Thanks!

**Comment by stevesbrain** [comment]
Tried it out - no success; instant drain!

**Comment by tasn** [comment]
I uploaded another version that completely removes task list updating, could you see if that fixes it? Maybe my tests to see if anything changed were wrong.

**Comment by stevesbrain** [comment]
Sure - just uploaded to same link as before?

**Comment by tasn** [comment]
Yeah.

**Comment by tasn** [comment]
Sorry, only now though, it didn't upload for some reason.

**Comment by stevesbrain** [comment]
Same problem sorry!

**Comment by tasn** [comment]
@dmfs: I just removed the code that updates tasklist name/color/etc, and he's still experiencing the same problem. Let's wait for your monitoring code to get a better idea of what's going on.

**Comment by stevesbrain** [comment]
As an illustrative image, please see the following:

![Screenshot](https://cloud001.tuxcon.com/file/public-downloads/screenshots/battery.png)

So prior to first spike, I had EteSync already installed, and I installed OpenTasks. I create a new task journal, and remove the old one. I manually sync. Then, I switch to OpenTasks, create a new entry, back to EteSync, and manually switch. Then, send EteSync to the background, modify the task in OpenTasks, then wait a short time (screen off or on, doesn't matter - off in this case). Then you see the first spike. That spike remained until I open EteSync and manually force a sync, at which point you see that second drop back to normal.

**Comment by dmfs** [comment]
@stevesbrain can you run the ContentObserver app again whis this new version? I'll get back to you with the verbose OpenTasks app in a moment.

**Comment by dmfs** [comment]
> Isn't OpenTasks better suited to check if something have changed and only trigger a "changed" for the UI applications if that's the case?

Yes, OpenTasks can easily check that and suppress any actions if nothing needs to be done. That's going to be improved by dmfs/opentasks#836.

However, such kind of optimization should be performed at the earliest stage possible to avoid unnecessary overhead in the lower levels or on the transport layer. Even if we handle this in OpenTasks, it would still result in Binder transactions to transfer the data and CPU cycles to read the database and compare the values. At that point your sync code already knows that nothing has changed and can easily avoid the overhead.

Consider this, ideally the server only sends you the tasks which have been changed and don't leave it to the client to figure that out. Of course the client can do that, but it's just more efficient doing this as early as possible in the chain. Same thing.

Anyway, I'm curious why this sync loop still exists. I'd expect the loop to be interrupted if you don't update the task list when nothing has changed.

**Comment by dmfs** [comment]
@stevesbrain please try this version: [opentasks-debug.zip](https://github.com/etesync/android/files/3560204/opentasks-debug.zip)
Note the version is signed with our release key. You'll have to uninstall any other version signed by a different key. The verbose logs go to logcat with the tag `OpenTasks`. I presume you know how to read these.

**Comment by tasn** [comment]
@dmfs: I agree we also need to fix it. :)

**Comment by stevesbrain** [comment]
@dmfs this is the result of ContentObserver from the new EteSync version, before the verbose OpenTasks has been installed:

`
20190831T082652 content://org.dmfs.tasks/instances
20190831T082652 content://org.dmfs.tasks/tasks/1
20190831T082652 content://org.dmfs.tasks/tasks
20190831T082729 content://org.dmfs.tasks/instances
20190831T082729 content://org.dmfs.tasks/tasks/1
20190831T082729 content://org.dmfs.tasks/tasks
`

I'll next install the verbose OpenTasks and get back to you with those results. Thanks!

**Comment by stevesbrain** [comment]
@dmfs This is the logcat entries:

`
08-31 18:55:59.248  1168  5962 I ActivityManager: START u0 {act=android.intent.action.VIEW dat=content://com.android.chrome.FileProvider/downloads/opentasks-debug.zip typ=application/zip flg=0x10000003 cmp=com.speedsoftware.explorer/.Explorer} from uid 10092
08-31 18:56:12.706  1168  8931 I ActivityManager: START u0 {act=android.intent.action.VIEW dat=content://com.speedsoftware.explorer.fileprovider/root/storage/emulated/0/SpeedSoftware/Extracted/opentasks-debug.apk typ=application/vnd.android.package-archive flg=0x3 cmp=com.google.android.packageinstaller/com.android.packageinstaller.InstallStart} from uid 10162
08-31 18:56:12.818  1168  8931 I ActivityManager: START u0 {act=android.intent.action.VIEW dat=content://com.speedsoftware.explorer.fileprovider/root/storage/emulated/0/SpeedSoftware/Extracted/opentasks-debug.apk typ=application/vnd.android.package-archive flg=0x2000000 cmp=com.google.android.packageinstaller/com.android.packageinstaller.InstallStaging (has extras)} from uid 10025
08-31 18:57:23.414 12161 13228 V OpenTasks: Delete on content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
08-31 18:57:23.420 12161 13228 V OpenTasks: notifyChange on content://org.dmfs.tasks/instances, requesting sync to network: false
08-31 18:57:23.421 12161 13228 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasks, requesting sync to network: false
08-31 18:57:23.423 12161 13228 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true, requesting sync to network: false
08-31 18:57:23.426 12161 12206 V OpenTasks: Update on content://org.dmfs.tasks/content_operation/UPDATE_NOTIFICATION_ALARM, Values:
08-31 18:57:23.436 12161 13228 V OpenTasks: Delete on content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
08-31 18:57:23.441 12161 13228 V OpenTasks: notifyChange on content://org.dmfs.tasks/instances, requesting sync to network: false
08-31 18:57:23.442 12161 13228 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasks, requesting sync to network: false
08-31 18:57:23.443 12161 13228 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true, requesting sync to network: false
08-31 18:57:23.446 12161 12206 V OpenTasks: Update on content://org.dmfs.tasks/content_operation/UPDATE_NOTIFICATION_ALARM, Values:
08-31 18:57:23.461 12161 12174 V OpenTasks: Insert on content://org.dmfs.tasks/tasklists?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true, Values: account_type=com.etesync.syncadapter _sync_id=REDACT46f50909543ceef26ab9c1e09c6220faecba4a27 visible=1 list_owner=ete@domain.com sync_enabled=1 account_name=ete@domain.com list_color=-7617718 list_name=One  list_access_level=0
08-31 18:57:23.464 12161 12174 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasklists?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true, requesting sync to network: false
08-31 18:57:23.466 12161 12174 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasklists/5, requesting sync to network: false
08-31 18:57:23.468 12161 12206 V OpenTasks: Update on content://org.dmfs.tasks/content_operation/UPDATE_NOTIFICATION_ALARM, Values:
08-31 18:57:31.717 12161 12161 V OpenTasks: Insert on content://org.dmfs.tasks/tasks, Values: list_id=5 title=Newb
08-31 18:57:31.732 12161 12161 V OpenTasks: notifyChange on content://org.dmfs.tasks/instances, requesting sync to network: true
08-31 18:57:31.735 12161 12161 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasks/1, requesting sync to network: true
08-31 18:57:31.736 12161 12161 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasks, requesting sync to network: true
08-31 18:57:31.740 12161 12206 V OpenTasks: Update on content://org.dmfs.tasks/content_operation/UPDATE_NOTIFICATION_ALARM, Values:
08-31 18:57:38.635 12161 13234 V OpenTasks: Update on content://org.dmfs.tasks/tasks/1?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true, Values: sync2=044866c1-f1a5-REDACTED _sync_id=044866c1-f1a5-REDACTED
08-31 18:57:39.324 12161 13234 V OpenTasks: Update on content://org.dmfs.tasks/tasks/1?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true, Values: _dirty=0 sync3=0 sync1=044866c1-f1a5-REDACTED
08-31 18:57:58.002 12161 12161 V OpenTasks: Update on content://org.dmfs.tasks/tasks/1, Values: description=
08-31 18:57:58.002 12161 12161 V OpenTasks: [ ] Djjsv
08-31 18:57:58.002 12161 12161 V OpenTasks: [ ] Cncnsxcv percent_complete=0
08-31 18:57:58.018 12161 12161 V OpenTasks: notifyChange on content://org.dmfs.tasks/instances, requesting sync to network: true
08-31 18:57:58.021 12161 12161 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasks/1, requesting sync to network: true
08-31 18:57:58.023 12161 12161 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasks, requesting sync to network: true
08-31 18:57:58.026 12161 12206 V OpenTasks: Update on content://org.dmfs.tasks/content_operation/UPDATE_NOTIFICATION_ALARM, Values:
08-31 18:58:42.781 12161 12174 V OpenTasks: Update on content://org.dmfs.tasks/tasks/1?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true, Values: _dirty=0 sync3=1 sync1=044866c1-f1a5-REDACTED
`

**Comment by stevesbrain** [comment]
@tasn I presume you've seen similar things in the debug log, but I saw many InterruptedIOException's for EteSync in the logcat logs. If debug logs didn't show those, let me know and I'll send them through too!

**Comment by tasn** [comment]
I saw them. I believe they are because OpenTasks is triggering syncs that's interrupting the existing syncs.

**Comment by dmfs** [comment]
Ok, here is the log with newlines, grouped and commented:
```
// removal of task list 3 by sync adapter -> no sync triggered
08-31 18:57:23.414 12161 13228 V OpenTasks: Delete on content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
08-31 18:57:23.420 12161 13228 V OpenTasks: notifyChange on content://org.dmfs.tasks/instances, requesting sync to network: false
08-31 18:57:23.421 12161 13228 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasks, requesting sync to network: false
08-31 18:57:23.423 12161 13228 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasklists/3?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true, requesting sync to network: false

// internal operation to trigger an update of the alarm timer -> no notification
08-31 18:57:23.426 12161 12206 V OpenTasks: Update on content://org.dmfs.tasks/content_operation/UPDATE_NOTIFICATION_ALARM, Values:

// removal of task list 4 by sync adapter -> no sync triggered
08-31 18:57:23.436 12161 13228 V OpenTasks: Delete on content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true
08-31 18:57:23.441 12161 13228 V OpenTasks: notifyChange on content://org.dmfs.tasks/instances, requesting sync to network: false
08-31 18:57:23.442 12161 13228 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasks, requesting sync to network: false
08-31 18:57:23.443 12161 13228 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasklists/4?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true, requesting sync to network: false

// internal operation to trigger an update of the alarm timer -> no notification
08-31 18:57:23.446 12161 12206 V OpenTasks: Update on content://org.dmfs.tasks/content_operation/UPDATE_NOTIFICATION_ALARM, Values:

// creation of a new task list ""One" by the sync adapter -> no sync triggered
08-31 18:57:23.461 12161 12174 V OpenTasks: Insert on content://org.dmfs.tasks/tasklists?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true, Values: account_type=com.etesync.syncadapter _sync_id=REDACT46f50909543ceef26ab9c1e09c6220faecba4a27 visible=1 list_owner=ete@domain.com sync_enabled=1 account_name=ete@domain.com list_color=-7617718 list_name=One list_access_level=0
08-31 18:57:23.464 12161 12174 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasklists?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true, requesting sync to network: false
08-31 18:57:23.466 12161 12174 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasklists/5, requesting sync to network: false

// internal operation to trigger an update of the alarm timer -> no notification
08-31 18:57:23.468 12161 12206 V OpenTasks: Update on content://org.dmfs.tasks/content_operation/UPDATE_NOTIFICATION_ALARM, Values:

// ... Task "Newb" is created in OpenTasks ...

// task "Newb" created on the new task list by non-sync adapter -> sync triggered
08-31 18:57:31.717 12161 12161 V OpenTasks: Insert on content://org.dmfs.tasks/tasks, Values: list_id=5 title=Newb
08-31 18:57:31.732 12161 12161 V OpenTasks: notifyChange on content://org.dmfs.tasks/instances, requesting sync to network: true
08-31 18:57:31.735 12161 12161 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasks/1, requesting sync to network: true
08-31 18:57:31.736 12161 12161 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasks, requesting sync to network: true

// internal operation to trigger an update of the alarm timer -> no notification
08-31 18:57:31.740 12161 12206 V OpenTasks: Update on content://org.dmfs.tasks/content_operation/UPDATE_NOTIFICATION_ALARM, Values:

// ... sync starts here ...

// update of sync columns of task no 1 by sync adapter -> no notification
08-31 18:57:38.635 12161 13234 V OpenTasks: Update on content://org.dmfs.tasks/tasks/1?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true, Values: sync2=044866c1-f1a5-REDACTED _sync_id=044866c1-f1a5-REDACTED

// update of sync columns of task no 1 by sync adapter -> no notification
08-31 18:57:39.324 12161 13234 V OpenTasks: Update on content://org.dmfs.tasks/tasks/1?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true, Values: _dirty=0 sync3=0 sync1=044866c1-f1a5-REDACTED

// ... task "Newb" is edited in OpenTasks ...

// update of value columns by non-sync adapter -> sync triggered
08-31 18:57:58.002 12161 12161 V OpenTasks: Update on content://org.dmfs.tasks/tasks/1, Values: description=
08-31 18:57:58.002 12161 12161 V OpenTasks: [ ] Djjsv
08-31 18:57:58.002 12161 12161 V OpenTasks: [ ] Cncnsxcv percent_complete=0
08-31 18:57:58.018 12161 12161 V OpenTasks: notifyChange on content://org.dmfs.tasks/instances, requesting sync to network: true
08-31 18:57:58.021 12161 12161 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasks/1, requesting sync to network: true
08-31 18:57:58.023 12161 12161 V OpenTasks: notifyChange on content://org.dmfs.tasks/tasks, requesting sync to network: true

// internal operation to trigger an update of the alarm timer -> no notification
08-31 18:57:58.026 12161 12206 V OpenTasks: Update on content://org.dmfs.tasks/content_operation/UPDATE_NOTIFICATION_ALARM, Values:

// ... sync starts here ...

// update of sync columns of task no 1 by sync adapter -> no notification
08-31 18:58:42.781 12161 12174 V OpenTasks: Update on content://org.dmfs.tasks/tasks/1?account_name=ete%40domain.com&account_type=com.etesync.syncadapter&caller_is_syncadapter=true, Values: _dirty=0 sync3=1 sync1=044866c1-f1a5-REDACTED
```
According to this log, OpenTasks is doing everything right. That's exactly the intended behavior. There were only two operations which trigger a sync are creating an updating the task in OpenTasks.

To me it looks like the loop is in the sync adapter itself. This might also explain the `InterruptedExceptions`. In my early days I saw them in my sync apps too. Back then Android also tried to kill the sync thread if the sync adapter didn't cause any network traffic for about 90 seconds (I can't remember the exact duration). I guess that's meant to avoid keeping stalled sync adapter running and I assume it's still in there.

**Comment by tasn** [comment]
This is not "Interrupted" it's "InterruptedIO". Also, it's exactly the same code that powers the contacts and calendar sync which has no issues. I don't know why it goes in a loop then. I'll have to continue investigating.

Thanks for your help @dmfs, I'll ping you again if I find more concrete evidence of OpenTasks misbehaving.

**Comment by stevesbrain** [comment]
Funny enough, Android 10 seems to have resolved this issue :/ will keep an eye on it to see if it's really gone - but so far so good!

**Comment by tasn** [comment]
I'm so sorry, I completely forgot to follow up on this one! Glad to see it's fixed, please report back in a few hours / days to let us know if it came back or gone for good. Thanks!

**Comment by stevesbrain** [comment]
Was going well - had one day where it appeared to get a bit warm but battery wasn't ridiculous. Uninstalled and it got cooler - so gonna test again in a couple of days and see if that was just coincidence or not :)

**Comment by tasn** [comment]
OK, let me know how it goes. I hope it's not back. :|

**Comment by dmfs** [comment]
FYI, I've started to roll out OpenTasks 1.2.1 which contains a [change](https://github.com/dmfs/opentasks/commit/f7d1f5fa729de9aa07b80faea354914eb21ae5a3#diff-a2d9992674dba3695927f124d604a7b4R1112) to eliminate any notifications and sync triggers for update operations which don't change any values. Not sure when F-Droid will pick up the new version though.

**Comment by tasn** [comment]
Any updates on this one @stevesbrain? I think I've just seen another user suffering from this. Not sure though. :(

**Comment by stevesbrain** [comment]
Sorry, have had a fairly busy week since a role change at work. Got a week off now though, so will take another look!

**Comment by tasn** [comment]
No worries! Please let us know once you know.

**Comment by tasn** [comment]
Hey, what's up? Just following up... :)

**Comment by stevesbrain** [comment]
Hey @tasn, sorry for delay - many "IRL" things cropped up. Got a new 4XL for what it's worth, and it seems to be working fine on that so far!

**Comment by tasn** [comment]
Great! So I'm closing this one for now. If anyone encounters it: please reopen.
